lib/pbio/drv/reset/reset_ev3.c

```diff
@@ -23,17 +30,31 @@ void pbdrv_reset_init(void) {
 void pbdrv_reset(pbdrv_reset_action_t action) {
     for (;;) {
         switch (action) {
-            case PBDRV_RESET_ACTION_RESET_IN_UPDATE_MODE:
-                // TODO
+            case PBDRV_RESET_ACTION_POWER_OFF:
+                pbdrv_reset_power_off();
                 break;
-            // TODO: implement case PBDRV_RESET_ACTION_RESET
+            case PBDRV_RESET_ACTION_RESET_IN_UPDATE_MODE:
+                *(volatile uint32_t *)BOOTLOADER_UPDATE_MODE_ADDR = BOOTLOADER_UPDATE_MODE_VALUE;
+            // Fall through
```

Comment:
```suggestion
            __attribute__((fallthrough));
```

Probably worth starting a `pbio_compiler_*` header that has macros for this, the memory barrier inline assembly, and other common things we use to give hints to the compiler.

Reply:
I've chosen `lib/pbio/include/pbdrv/compiler.h` since most of these are very low-level tasks most likely to be found in the driver layer.